include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -767,6 +767,22 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+template <typename _Tp>
+union __lazy_ctor_storage
```

Comment:
Would the location of this union be better suited as a general utility in `pstl/utils.h` as it is not SYCL specific?